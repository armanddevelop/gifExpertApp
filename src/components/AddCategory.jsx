import { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const handleChange = (e) => {
    setInputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    inputValue.trim().length > 3 &&
      setCategories((category) => [inputValue, ...category]);
    setInputValue("");
  };
  return (
    <form onSubmit={handleSubmit} noValidate>
      <input
        placeholder="Search for a Gif"
        type="text"
        value={inputValue}
        onChange={(e) => handleChange(e)}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
