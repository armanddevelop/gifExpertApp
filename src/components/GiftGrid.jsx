import { useFetchGift } from "../Hooks/useFetchGif";
import { GiftGridItem } from ".";
import PropTypes from "prop-types";
export const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetchGift(category);
  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && (
        <p className="animate__animated animate__flash">Cargando.....</p>
      )}
      <div className="card-grid">
        {images.map((image) => (
          <GiftGridItem key={image.id} img={image} />
        ))}
      </div>
    </>
  );
};
GiftGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
