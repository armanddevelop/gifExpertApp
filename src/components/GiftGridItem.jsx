export const GiftGridItem = ({ img }) => {
  const { title, url, username } = img;
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={username} />
      <p>{title === "" ? "no title" : title}</p>
    </div>
  );
};
