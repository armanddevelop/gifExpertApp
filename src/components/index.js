export { GiftGridItem } from "./GiftGridItem";
export { GiftGrid } from "./GiftGrid";
export { AddCategory } from "./AddCategory";
