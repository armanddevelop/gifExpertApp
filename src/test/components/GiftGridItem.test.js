import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { GiftGridItem } from "../../components";

describe("Test for GiftGridItem component", () => {
  const propsGiftItem = {
    title: "No image",
    url: "http://wwww.hola.com",
    username: "licha",
  };
  const wrapper = shallow(<GiftGridItem img={propsGiftItem} />);
  it("snapshot for the GiftGridItem component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("Should show title in the component", () => {
    const paragraph = wrapper.find("p");
    expect(paragraph.text().trim()).toBe(propsGiftItem.title);
  });
  it("Should show no title in the component", () => {
    const propsGiftItem = {
      title: "",
      url: "http://wwww.hola.com",
      username: "licha",
    };
    const wrapper = shallow(<GiftGridItem img={propsGiftItem} />);
    const paragraph = wrapper.find("p");
    expect(paragraph.text().trim()).toBe("no title");
  });
  it("Should show src and alt in the img element", () => {
    const img = wrapper.find("img");
    const { src, alt } = img.props();
    expect(src).toBe(propsGiftItem.url);
    expect(alt).toBe(propsGiftItem.username);
  });
  it("Should have className animated__fadeIn in component", () => {
    const card = wrapper.find("div");
    expect(card.prop("className")).toMatch(/animate__fadeIn/);
  });
});
