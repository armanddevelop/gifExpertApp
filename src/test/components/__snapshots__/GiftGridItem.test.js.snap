// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for GiftGridItem component snapshot for the GiftGridItem component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="licha"
    src="http://wwww.hola.com"
  />
  <p>
    No image
  </p>
</div>
`;
